relay-kafka/src/config.rs

```diff
@@ -82,18 +82,20 @@ impl KafkaTopic {
 }
 
 macro_rules! define_topic_assignments {
-    ($struct_name:ident {
-        $($field_name:ident : ($kafka_topic:path, $default_topic: literal, $doc: literal)),* $(,)? }) => {
-
+    ($($field_name:ident : ($kafka_topic:path, $default_topic:literal, $doc:literal)),* $(,)?) => {
         /// Configuration for topics.
         #[derive(Serialize, Deserialize, Debug)]
         #[serde(default)]
         pub struct TopicAssignments {
             $(
                 #[serde(alias = $default_topic)]
-                #[doc=$doc]
+                #[doc = $doc]
                 pub $field_name: TopicAssignment,
             )*
+
+            /// Additional topic assignments configured but currently unused by this Relay instance.
+            #[serde(flatten)]
+            pub unused: BTreeMap<String, TopicAssignment>,
```

Comment:
Should we deny unknown fields instead? https://serde.rs/container-attrs.html#deny_unknown_fields

Reply:
We could, see the PR description, deny unknown adds some behaviour we maybe don't want.

Reply:
Thanks, I should have read the description. So the error is more like a warning, maybe we should start reporting warnings to sentry.

We could theoretically log the error in a `deserialize_with` function on `unused` (not sure if that's better, just brainstorming here).